geo/src/algorithm/relate/geomgraph/intersection_matrix.rs

```diff
@@ -268,6 +278,225 @@ impl IntersectionMatrix {
+    /// # Notes
+    /// - This predicate is **symmetrice**
+    #[allow(clippy::nonminimal_bool)]
+    pub fn is_touches(&self) -> bool {
+        // [FT*******]
+        self.0[CoordPos::Inside][CoordPos::Inside] == Dimensions::Empty
+        && self.0[CoordPos::Inside][CoordPos::OnBoundary] != Dimensions::Empty ||
+        // [F**T*****]
+        self.0[CoordPos::Inside][CoordPos::Inside] == Dimensions::Empty
+        && self.0[CoordPos::OnBoundary][CoordPos::Inside] != Dimensions::Empty ||
+        // [F***T****]
+        self.0[CoordPos::Inside][CoordPos::Inside] == Dimensions::Empty
+        && self.0[CoordPos::OnBoundary][CoordPos::OnBoundary] != Dimensions::Empty
+    }
+
+    /// Compares two geometry objects and returns `true` if their intersection "spatially crosses";
+    /// that is, the geometries have some, but not all interior points in common
+    ///
+    /// ```
+    /// use geo_types::{LineString, line_string, polygon};
+    /// use geo::relate::Relate;
+    ///
+    /// let line_string: LineString = line_string![(x: 85.0, y: 194.0), (x: 162.0, y: 135.0)];
+    /// let poly = polygon![
+    ///     (x: 125., y: 179.),
+    ///     (x: 110., y: 150.),
+    ///     (x: 160., y: 160.),
+    ///     (x: 125., y: 179.),
+    /// ];
+    ///
+    /// let intersection = line_string.relate(&poly);
+    /// assert_eq!(intersection.is_crosses(), true);
+    /// ```
+    /// # Notes
+    /// - If any of the following do not hold, the function will return `false`:
+    ///     - The intersection of the interiors of the geometries must be non-empty
+    ///     - The intersection must have dimension less than the maximum dimension of the two input geometries
+    ///     - The intersection of the two geometries must not equal either geometry
+    /// - This predicate is **symmetric**
+    pub fn is_crosses(&self) -> bool {
+        let dims_a = self.0[CoordPos::Inside][CoordPos::Inside]
+            .max(self.0[CoordPos::Inside][CoordPos::OnBoundary])
+            .max(self.0[CoordPos::Inside][CoordPos::Outside]);
+
+        let dims_b = self.0[CoordPos::Inside][CoordPos::Inside]
+            .max(self.0[CoordPos::OnBoundary][CoordPos::Inside])
+            .max(self.0[CoordPos::Outside][CoordPos::Inside]);
+
+        if dims_a != dims_b {
+            // input geom dimensions differ : proceed to calculating relationships between dimensions
+            let ii = self.0[CoordPos::Inside][CoordPos::Inside];
+            let ie = self.0[CoordPos::Inside][CoordPos::Outside];
+            let ei = self.0[CoordPos::Outside][CoordPos::Inside];
+            let a_lessthan_b = ii == ie;
+            let a_greaterthan_b = ii == ei;
+            // something has gone horribly wrong
+            debug_assert!(a_lessthan_b != a_greaterthan_b,
+                "Input geometry dimensions cannot simultaneously be greater than and less than each other. Are your geometries valid?"
+            );
+            if a_lessthan_b {
```

Comment:
```suggestion
            if dims_a < dims_b {
```

This can be simplified now that we have `dims_` - I think it'll be easier to track than computing the ii/ie/ei.

Additionally, it might be really slick to have all the cases in a match, but I leave that to your judgement.

(Introducing a `match` might be overly baroque, but we should at least get rid of `a_less_than_b`)

Reply:
I do like a match…